util/topgen/merge.py

```diff
@@ -107,7 +107,8 @@ def elaborate_instance(instance, block: IpBlock):
         param_prefixes = ["Sec", "RndCnst", "MemSize"]
         name_top = cc_mod_name + param.name
         for prefix in param_prefixes:
-            if param.name.lower().startswith(prefix.lower()):
+            if param.name.lower().startswith(
+                    prefix.lower()) and not param.name.startswith("Secure"):
```

Comment:
This is going to go wrong when someone adds a "SecludedMode" parameter or something :-) Could we make it do the following?
1. If `param.name.lower()` doesn't start with `prefix.lower()` then skip.
2. Otherwise, check that `param.name` starts with `prefix`. If not, that's an error.
3. Check that `param.name` isn't equal to `prefix`. That would be an error too.
3. Check whether the following letter is capitalised. If so, we really have "SecFoo". If not, we're in the "SecureFoo" / "SecludedFoo" case, and can ignore this parameter.

Reply:
Thanks for your reviews @rswarbrick and @imphil. I've changed this as you suggested. We now only move the prefix if its really a prefix, meaning it's followed by a capital letter.

Regarding why we are doing this: there are some parameters that we absolutely want to override in the final top level (RndCnst) and some that we absolutely don't want to override (Sec). Making sure the prefix remains a prefix in the top level (as topgen prepends the name of the instance to the parameter) is useful to not loose these parameters out of sight.